plutus-tx/src/PlutusTx/Code.hs

```diff
@@ -149,7 +153,8 @@ getPirNoAnn
     => CompiledCodeIn uni fun a -> Maybe (PIR.Program PIR.TyName PIR.Name uni fun ())
 getPirNoAnn = fmap void . getPir
 
-getCovIdx :: CompiledCodeIn uni fun a -> CoverageIndex
+-- | Will throw an error if the `CoverageIndex` is present but cannot be deserialised.
+getCovIdx :: CompiledCodeIn uni fun a -> Maybe CoverageIndex
 getCovIdx wrapper = case wrapper of
-  SerializedCode _ _ idx   -> idx
+  SerializedCode _ _ idx   -> unsafeFromEither . unflat <$> idx
```

Comment:
The error emitted by `unsafeFromEither` is not telling what invariant was violated, its too generic.
I suggest to pattern match and throw a more explanatory error, e.g. "SerializedCode was expected to contain information about CodeCoverage because of ... but it doesn't"